prompts/sources.go

```diff
@@ -657,3 +683,75 @@ func configureRegistry(source *workflow.Source, orgSlug, workspaceSlug, sourceNa
 	source.Registry = registryEntry
 	return nil
 }
+
+type FormatType string
+
+const (
+	FormatJSON FormatType = "json"
+	FormatYAML FormatType = "yaml"
+)
+
+type blueprintRequest struct {
+	ID string `json:"id"`
+}
+
+type blueprintResponse struct {
+	ID        string     `json:"id"`
+	Spec      string     `json:"spec"`
+	CreatedAt time.Time  `json:"created_at"`
+	Format    FormatType `json:"format"`
+}
+
+var (
+	ErrMsgFailedToFetchBlueprint  = errors.New("failed to fetch sandbox session")
+	ErrMsgFailedToSaveBlueprint   = errors.New("failed to save sandbox session")
+	ErrMsgFailedToDecodeBlueprint = errors.New("failed to decode sandbox session")
+)
+
+func fetchAndSaveBlueprint(ctx context.Context, blueprintID string) (string, error) {
+	baseURL := "https://api.speakeasy.com"
+	url := fmt.Sprintf("%s/v1/schema_store", baseURL)
+
+	var reqBody blueprintRequest
+	reqBody.ID = blueprintID
+	body, err := json.Marshal(reqBody)
+	if err != nil {
+		return "", ErrMsgFailedToDecodeBlueprint
+	}
+	req, err := http.NewRequest(http.MethodGet, url, bytes.NewBuffer(body))
+	if err != nil {
+		return "", ErrMsgFailedToFetchBlueprint
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", ErrMsgFailedToFetchBlueprint
+	}
+
+	var respBody blueprintResponse
+	err = json.NewDecoder(resp.Body).Decode(&respBody)
+	if err != nil {
+		return "", err
+	}
+
+	tempDir := os.TempDir()
```

Comment:
Why do we download to a temp location?

Reply:
https://github.com/speakeasy-api/speakeasy/pull/1291#discussion_r1983146063